src/classes/CON_PrimaryAffls_TDTM.cls

```diff
@@ -51,151 +51,200 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
+        List<Affiliation__c> affiliationsToInsert = this.locateContactAffiliationService().getAffiliationsToCreate(newContactById.values(), primaryAffiliationModels);
+
+        if (affiliationsToInsert.size() == 0) {
+            this.unsetInsertRecursion();
+            return;
+        }
+
+        this.setAffiliationUpdateRecursion();
+
+        DmlWrapper dmlWrapper = new DmlWrapper();
+        dmlWrapper.objectsToInsert.addAll((List<SObject>)affiliationsToInsert);
+        TDTM_TriggerHandler.processDML(dmlWrapper, true);
+        this.resetAffiliationUpdateRecursion();
+        this.unsetInsertRecursion();
+
     }
 
-    private List<Affiliation__c> makeOtherAfflsNotPrimary(Map<ID, List<ID>> changedConToAccsMap) {
-        //Find all the affiliations to make non-primary.
-        List<Affiliation__c> afflsToMakeNonPrimary = new List<Affiliation__c>();
+    /*******************************************************************************************************
+    * @description Returns if Contact Deletion Prevention custom setting is not enabled or
+    * Passes a map of contacts By Id to SRVC_Contact_PreventDeletion and returns a List of Contacts that have
+    * been prevented for deletion
+    * @param oldSObjectList list of sObject records that are being deleted
+    * @return List<Contact>
+    ********************************************************************************************************/
+    @TestVisible
+    private void handleAfterUpdate(List<SObject> oldSObjectList, List<SObject> newSObjectList) {
+
+        if (this.isUpdateRecursion() == true) {
+            return;
+        }
         
-        //Initially we get all possible affiliations from those contacts.
-        List<Affiliation__c> afflsAll = [select Contact__c, Account__c, Primary__c from Affiliation__c where Contact__c in :changedConToAccsMap.keySet()];
-        UTIL_Debug.debug('****All possible affls to make non-primary: ' + JSON.serializePretty(afflsAll));
         
-        //Then we refine the search - changedConToAccsMap should contain only the Acc IDs of those key Affl lookups that have been modified.
-        for (Affiliation__c affl : afflsAll) {
-            List<ID> contactAccIDs = changedConToAccsMap.get(affl.Contact__c);
-            for (ID changedAccID : contactAccIDs) {
-                if (affl.Account__c == changedAccID) {
-                    affl.Primary__c = false;
-                    UTIL_Debug.debug('****Making affl ' + affl.ID + ' non-primary.');
-                    afflsToMakeNonPrimary.add(affl);
-                }
+        setUpdateRecursion(); 
+
+        Map<Id, Contact> oldContactById = new Map<Id, Contact>((List<Contact>)oldSObjectList);
+        Map<Id, Contact> newContactById = new Map<Id, Contact>((List<Contact>)newSObjectList);
+
+        List<MODL_PrimaryAfflMapper> primaryAffiliationModels = this.locateAffiliationSettingMapper().getModelsFromSettings();
```

Comment:
It looks like this is used by both afterInsert and afterUpdate. Should we consider extract it so that it can be shared?

Reply:
@xinnanlisf This is a deliberate choice so we can swap out the mapper with an interface in the future without incurring a necessary update in this method.